backend/src/services/session.ts

```diff
@@ -0,0 +1,103 @@
+import { Types } from 'mongoose';
+import { InternalError, ServiceError } from "../errors";
+import { Session, SessionDoc } from '../models';
+
+export async function getUpcomingSession(userId: string, role: 'mentor' | 'mentee'): Promise<Types.ObjectId | null> {
+  const now = new Date();
+
+  let matchField: string;
+
+  if (role === 'mentor') {
+    matchField = 'mentorId';
+  } else if (role === 'mentee') {
+    matchField = 'menteeId';
+  } else {
+    throw ServiceError.INVALID_ROLE_WAS_FOUND
+  }
+
+  try {
+    const upcomingSession = await Session.findOne({
+      startTime: { $gt: now },
+      [matchField]: new Types.ObjectId(userId),
+    }).sort({ startTime: 1 });
+
+    return upcomingSession?._id;
+  } catch (error) {
+    throw InternalError.ERROR_FINDING_UPCOMING_SESSION
+  }
+}
+
+export async function getLastSession(userId: string, role: 'mentor' | 'mentee'): Promise<Types.ObjectId | null> {
+  const now = new Date();
+
+  let matchField: string;
+
+  if (role === 'mentor') {
+    matchField = 'mentorId';
+  } else if (role === 'mentee') {
+    matchField = 'menteeId';
+  } else {
+    throw ServiceError.INVALID_ROLE_WAS_FOUND;
+  }
+
+  try {
+    const lastSession = await Session.findOne({
+      endTime: { $lt: now },
+      [matchField]: new Types.ObjectId(userId),
+    }).sort({ endTime: -1 });
+
+    return lastSession?._id;
+  } catch (error) {
+    throw InternalError.ERROR_FINDING_PAST_SESSION;
+  }
+}
+
+/**
+ * Function takes in startTime and endTime date objects and formats it to give:
+ * [
+ *    "Thursday, April 27, 2023" OR "Thursday, April 27, 2023 - Friday, April 28, 2023",
+ *    "5/22"
+ *    "2:00 PM"
+ *    "2:30 PM"
+ * ]
+ */
+export function formatDateTimeRange(startTime: Date, endTime: Date): [string, string, string, string] {
```

Comment:
Date formatting taken care of on backend so that frontend logic is simple
 
fyi @HarshGurnani